src/test/java/org/jabref/logic/importer/fetcher/GrobidCitationFetcherTest.java

```diff
@@ -0,0 +1,206 @@
+        parseUsingGrobid.setAccessible(true);
+        assertEquals(defaultReturnValue, parseUsingGrobid.invoke(grobidCitationFetcher, invalidInput1));
+        assertEquals(defaultReturnValue, parseUsingGrobid.invoke(grobidCitationFetcher, invalidInput2));
+    }
+
+    /**
+     * Tests if the Optional<BibEntry> is filled correctly with the extracted fields from bibtex string..
+     */
+    @Test
+    public void grobidParseBibToBibEntryTest() throws IllegalAccessException, InvocationTargetException, NoSuchMethodException {
+        Method parseBibToBibEntry = GrobidCitationFetcher.class.getDeclaredMethod("parseBibToBibEntry", String.class);
+        parseBibToBibEntry.setAccessible(true);
+        Optional<BibEntry> bibEntry = (Optional<BibEntry>)parseBibToBibEntry.invoke(grobidCitationFetcher, example1AsBibtex);
+        assertTrue(bibEntry.isPresent());
+        assertEquals(bibEntry.get().getField(StandardField.AUTHOR).get(), "T Derwing and M Rossiter and M Munro");
+        assertEquals(bibEntry.get().getField(StandardField.TITLE).get(), "Teaching native speakers to listen to foreign-accented speech");
+        assertEquals(bibEntry.get().getField(StandardField.JOURNAL).get(), "Journal of Multilingual and Multicultural Development");
+        assertEquals(bibEntry.get().getField(StandardField.YEAR).get(), "2002");
+    }
+
+    /**
+     * Checks if parseBibToBibEntry creates a empty BibEntry on empty response
+     */
+    @Test
+    public void grobidParseBibToBibEntryReturnsEmptyOptionalWhenFailedTest() throws IllegalAccessException, NoSuchMethodException, InvocationTargetException {
+        Method parseBibToBibEntry = GrobidCitationFetcher.class.getDeclaredMethod("parseBibToBibEntry", String.class);
+        parseBibToBibEntry.setAccessible(true);
+        Optional<BibEntry> bibEntry = (Optional<BibEntry>)parseBibToBibEntry.invoke(grobidCitationFetcher, defaultReturnValue);
+        assertTrue(bibEntry.isPresent());
+        assertFalse(bibEntry.get().getField(StandardField.AUTHOR).isPresent());
+        assertFalse(bibEntry.get().getField(StandardField.TITLE).isPresent());
+        assertFalse(bibEntry.get().getField(StandardField.YEAR).isPresent());
+        assertFalse(bibEntry.get().getField(StandardField.JOURNAL).isPresent());
+    }
+
+    /**
+     * When the user runs the program without passing anything, no BibEntry is generated (The Optional should be empty)
+     */
+    @Test
+    public void grobidParseBibToBibEntryReturnsNoBibEntryWhenPassingInvalidInputTest() throws IllegalAccessException, NoSuchMethodException, InvocationTargetException {
+        Method parseBibToBibEntry = GrobidCitationFetcher.class.getDeclaredMethod("parseBibToBibEntry", String.class);
+        parseBibToBibEntry.setAccessible(true);
+        Optional<BibEntry> bibEntry = (Optional<BibEntry>)parseBibToBibEntry.invoke(grobidCitationFetcher, "");
```

Comment:
Please double check that **all team members** followed the workspace setup guide: https://jabref.readthedocs.io/en/latest/guidelines-for-setting-up-a-local-workspace/ - please import the JabRef code style and reformat the code! (Strg+Alt+L in IntelliJ)